addons/account/tests/test_account_analytic.py

```diff
@@ -211,6 +211,28 @@ def test_mandatory_plan_validation(self):
         invoice.action_post()
         self.assertEqual(invoice.state, 'posted')
 
+    def test_mandatory_plan_validation_mass_posting(self):
+        """
+        In case of mass posting, we should still check for mandatory analytic plans.
+        """
+        invoice1 = self.create_invoice(self.partner_a, self.product_a)
+        invoice2 = self.create_invoice(self.partner_b, self.product_a)
+        self.default_plan.write({
+            'applicability_ids': [Command.create({
+                'business_domain': 'invoice',
+                'product_categ_id': self.product_a.categ_id.id,
+                'applicability': 'mandatory',
+            })]
+        })
+
+        vam = self.env['validate.account.move'].create({'force_post': True})
+        with self.assertRaisesRegex(RedirectWarning, '100% analytic distribution.'):
+            vam.with_context({
+                'active_model': 'account.move',
+                'active_ids': [invoice1.id, invoice2.id],
+                'validate_analytic': True,
+            }).validate_move()
```

Comment:
We can test the behavior in both cases and avoid testing the implementation or the user message (which is rather volatile).

*(Note that the two `with` have different scopes, so they can't be joined)*

```suggestion
        for invoices in [invoice1, invoice1 | invoice2]:
            with self.subTest(invoices=invoices): 
                with self.assertRaises(Exception):
                    vam.with_context({
                        'active_model': 'account.move',
                        'active_ids': invoices.ids,
                        'validate_analytic': True,
                    }).validate_move()
                self.assertTrue('posted' not in invoices.mapped('state'))
```